src/SignalingClient.ts

```diff
@@ -158,6 +159,36 @@ export class SignalingClient extends EventEmitter {
         }
     }
 
+    public reconnect(maxAttempt = 3): void {
+        this.reconnection = true;
+        if (this.readyState === ReadyState.CLOSED) {
+            this.attemptReconnect(maxAttempt);
+        } else {
+            console.log('Invalid state to invoke reconnect from');
+        }
+    }
+
+    private attemptReconnect(maxAttempts: number, attempt = 0): void {
+        if (this.readyState === ReadyState.CLOSED) {
+            this.open();
+            // If reconnect fails
+            setTimeout(() => {
+                if (this.readyState !== ReadyState.OPEN) {
```

Comment:
Logic needed to check if `close` was already called.

Reply:
setTimeout returns a timer id, you can use clearTimeout(id) in `close` to stop the pending countdown as well